Python/errors.c

```diff
@@ -1040,6 +1053,14 @@ write_unraisable_exc_file(PyObject *exc_type, PyObject *exc_value,
             return -1;
         }
     }
+    else if (err_msg != NULL && err_msg != Py_None) {
```

Comment:
I think you can avoid duplication of the code.

Reply:
I prefer to have a single PyFile_WriteString(":\n", file) call in this code path, rather than trying to factorize the code too much, to reduce the risk of failure in the hook.